Libraries/LibGUI/Model.h

```diff
@@ -107,6 +108,7 @@ class Model : public RefCounted<Model> {
 
 private:
     HashTable<AbstractView*> m_views;
+    Vector<Function<void(unsigned)>> m_on_update;
```

Comment:
Instead of having both of these, how about this:

An abstract `ModelClient` class that you can inherit from, and each `Model` has a set of N clients.
When something interesting happens, we iterate through the set of clients and call a virtual on each one.
Both `AbstractView` and `SortingProxyModel` then inherit from `ModelClient` and we funnel everything through a uniform interface. :)
That also makes it a bit more generic and opens up for easy extensibility with more hooks.

Reply:
I made these changes in another merge request: #2761
I'm closing this one.